SCRUM-17:Get_Email_Important

Diego-Claros17 opened the pull request:
Validation of schema of email endpoint (GET)

Diego-Claros17 committed:
Schema validation for email

Validation of schema of email endpoint

FreddyFZ17 commented on src/assertions/assertion_schemas.py:
The assert_schema_email method follows the rules set by the team and is easy to understand.

AlisonGuzman2006 replied:
This schema validation for this endpoint is accepted, this method has to be called upon in it's test

FreddyFZ17 commented on src/resources/schemas/email.json:
This is the json schema that provides the assigned endpoint so I don't have any observations.

AlisonGuzman2006 replied:
The schema created for the endpoint, is accepted

FreddyFZ17 commented on tests/Correo_borrador/test_GetEmail.py:
The test_get_email_succed method successfully validates the API response when getting emails from the drafts tray

AlisonGuzman2006 replied:
the method name has a typo, it should be "succeed" instead of "succed". Other than this i have no other obsertvations.

FreddyFZ17 reviewed (dismissed):
I don't have any observations in general and when I ran it I had no problems and I saw that it integrated well into the development branch.

AlisonGuzman2006 reviewed (dismissed):
Other than a small typo. I have no other observations, I approve this request.

Diego-Claros17 committed:
SCRUM-17: Get-Email-Draft

Implemented Test Cases to verify the following scenarios:

  - Successful retrieval of email drafts with status code 200.
  - Unauthorized access with wrong basic auth credentials.
  - Unauthorized access without auth token secret in the headers.
  - Bad request response when Content-Type is set to text/plain.
  - Bad request response with additional unexpected parameters.
  - Successful response when search parameter is empty.
  - Bad request response with invalid search fields.
  - Successful response when maxSize parameter is empty.
  - Successful response with valid maxSize parameter.
  - Bad request response with invalid maxSize parameter.
  - Bad request response with invalid offset type.
  - Bad request response with invalid order type.

Added detailed print statements for debugging and validation:
  - Print request headers, URL, response status code, and response text for selected test cases.

Note: While most tests passed, a few tests returned unexpectedly.

Diego-Claros17 committed:
Merge branch 'development' into get-email-diego

Diego-Claros17 committed:
Conflicts resolved

Diego-Claros17 committed:
restructuring of the code according to the new structure

Diego-Claros17 committed:
5 new TCS were added

Diego-Claros17 committed:
Some refactions

Diego-Claros17 committed:
Merge branch 'development' into get-email-diego

Diego-Claros17 committed:
Merge branch 'development' into get-email-diego

Diego-Claros17 committed:
Test result reporting tool added (Allure) Comment on my failing tests and slight tweaks to my code

AlisonGuzman2006 reviewed (dismissed):
This change only contains the comments for the bug, therefore its aproved

AlyGASS committed:
the code has been reviewed and corrected

AlisonGuzman2006 reviewed (approved):
Aproved

FreddyFZ17 reviewed (approved):
ok
